reactioncommerce/reaction
# feat: UI for adding Storefront Urls inside Catalyst

Resolves #5150   
Impact: **minor**  
Type: **feature**

## Issue
The UI portion of #5144 was not completed at the time 5144 was done.

## Solution
This PR adds the UI for #5144.

Also added a `withPrimaryShop` HOC, which provides shop information. Currently, only fields I needed for this PR are provided, they should probably be added as needed going forward.

## Breaking changes
None

## Testing
1. Navigate to /operator/shop-settings
1. Try changing the various fields
1. Refresh the page to make sure they have saved
1. Go into the Shops collection of mongo, and look at the `storefrontUrls` object, and make sure they have saved

<img width="1152" alt="Not-found" src="https://user-images.githubusercontent.com/4482263/57483956-b7ef9400-725c-11e9-9690-a399ea3ebb9b.png">
